### ISSUE TITLE
Language of site navigation

### ISSUE THREAD

**Issue by slhenry** [w3c/wai-translations#14]
At start of issue, proposal is:
* Translations include the full site navigation, header, and footer.
* Navigation text is in the language of the linked page (that is, if the page is translated, that navigation link is in that language, and if not, it's in English).
* implemented in [mock-up](https://w3c.github.io/wai-translation-playground/people-use-web/).

See issues, pros, cons, and considerations in: [Language of site nav - or not include in translations](https://www.w3.org/wiki/WAI_Translations#Language_of_site_nav_-_or_not_include_in_translations)

The proposal and mock-up might change over the life of this issue. In your comment, quote or clarify what you are commenting on. :-)

**Comment by slhenry** [comment]
I notice in some i18n translated pages, the main nav is translated even though some linked pages are in English.
(For example, in <https://www.w3.org/International/getting-started/index.de> I click on "Über" and I get an English page.)

Personally I find that disconcerting and annoying. (<--fyi, @r12a )

Thus:
-1 for translated links going to English pages without indication.

**Comment by sharronrush** [comment]
My preference is to grey out or remove navigation links to English pages and show only live navigation to translated pages, Exception would be a link to the WAI home page.

**Comment by yatil** [comment]
* I am pretty sure that no-one advocates for having English pages in the navigation without indication.
* We could use a grey for the English nav items, and `(en)` or something to indicate the language for people who cannot perceive color.

**Comment by slhenry** [comment]
Re: Including only translated pages in the navigation
Here are relevant comments from Eric from e-mails (shared with permission) [_emphasis mine_]
> I think showing the whole navigation with gaps of language support _**shows our commitment to translation**_. When I get to a website that has a nice little German corner with a fraction of the information, _**I don’t feel included**_ and I know there is no commitment to making all content available.

> having language “ghettos” means that people don’t see the breadth of our content, it also means we don’t _**show a commitment to translations to the user**_. ([e-mail](https://lists.w3.org/Archives/Public/public-eo-archive/2018OctDec/0015.html))

**Comment by slhenry** [comment]
Re: All navigation links translated, or only the pages available in that language are translated and the others are in English?

**I would very much like to get more user perspectives (such as above) from people who have experience using translated web pages &mdash; particularly when only a small part of the website is translated, not all of the website.**

I'd like us to get some shared understanding around the reactions at the F2F to the navigation being mostly in English and a little French. Hopefully we can get more of that via phone next week. Input here welcome, too!

/me **_guesses_** at couple different perspectives
> When I'm skimming through navigation, I want to easily know if the linked page is in English or my language. It's easiest for me if the link itself matches the language of the page. If all the navigation is translated, it takes a more effort for me read the title in my language and then process that the pages is actually in English (from icon or gray or "(in English)" or whatever).

...

> When I come to a page that I think is in my language, yet much of what I first see (header, navigation) is in English, I think the page I'm on is not is really in my language after all. It's confusing.

**Comment by r12a** [comment]
This is a tricky area.

I don't think we should remove links to pages that aren't in the current language,  because 
1. people may speak English (or some other content negotiated language that is available) too, even though they prefer to read their own language when possible.
2. people could run the page through an online translation service, but they are unlikely to even know there is information there if there's no link.

I suspect that greying out links of this kind would make people think that they aren't active, so not sure that's a good idea either.

I think, if anything, putting something like (en) alongside might work, as long as it's _automatically applied_ when the current page is rendered (ie. requires javascript).  The reason for that is that a page may be translated at some point (maybe even the next day), and you don't then want people not clicking on the link because they think they won't be able to read it.

**Comment by slhenry** [comment]
Thanks for the input, @r12a. That is line with our latest proposals, which are partly influenced by user experiences and perspectives discussed in [30 Oct meeting](https://www.w3.org/2018/10/30-eo-minutes.html)

Current plan is to include all navigation and links.

**Comment by yatil** [comment]
> I think, if anything, putting something like (en) alongside might work, as long as it's automatically applied when the current page is rendered (ie. requires javascript). The reason for that is that a page may be translated at some point (maybe even the next day), and you don't then want people not clicking on the link because they think they won't be able to read it.

To clarify: It does not need JavaScript in our system. When a translation is available, we can detect it and change the navigation in place. We want to do some user testing.

**Comment by r12a** [comment]
Do you mean with server-side scripting used to server the page?

**Comment by yatil** [comment]
> Do you mean with server-side scripting used to server the page?

No, we render static HTML pages with a static site generator ([Jekyll](https://jekyllrb.com)). I don’t want to go into the technical details, because that is a tangent :-)
